pandas/core/indexes/base.py

```diff
@@ -541,10 +541,10 @@ def is_(self, other) -> bool:
         --------
         Index.identical : Works like ``Index.is_`` but also checks metadata.
         """
-        try:
-            return self._id is other._id
-        except AttributeError:
-            return False
+        if self is other:
+            return True
+        else:
+            return self._id is not None and self._id is getattr(other, "_id", Ellipsis)
```

Comment:
umm how is this equal to Ellipsis?

Reply:
I used Ellipsis instead of None, because `self._id` might be None, triggering a false Positive.

It's probably still too obscure, so I've made a more explicit version in a new commit.